snuba/web/db_query.py

```diff
@@ -243,15 +245,33 @@ def execute_query_with_rate_limits(
     stats: MutableMapping[str, Any],
     query_settings: MutableMapping[str, Any],
     robust: bool,
+    applied_quota: Optional[RateLimitStats],
 ) -> Result:
     # XXX: We should consider moving this that it applies to the logical query,
     # not the physical query.
     with RateLimitAggregator(
         request_settings.get_rate_limit_params()
     ) as rate_limit_stats_container:
-        stats.update(rate_limit_stats_container.to_dict())
         timer.mark("rate_limit")
 
+        if applied_quota is not None:
```

Comment:
We need to propagate the quota decision till here because here is where we set the max_threads parameters, that depends on project concurrent. 
I think we need something more abstract than the project rate limit to identify how much quota is being used. But that would not look like very different than a "number of concurrent query being executed on this quota currently). Probably just a matter of naming.